test/distributed/fsdp/test_fsdp_checkpoint.py

```diff
@@ -277,6 +279,116 @@ def test_basic_checkpoint_end_to_end(
         dist.barrier()
 
 
+class CheckpointModule(nn.Module):
```

Comment:
nit: Should we add a non-linearity (e.g. ReLU) to this model just to be safe for the gradient correctness check between 'with AC' and 'without AC'?

I do not think it matters for this case, but I just generally avoid the case where the model is equivalent to one affine transformation since I have had silent issues before.